Developer documentation for access handler (#60)
common-fate/granted-approvals: docs/access-handler/providers.md

## Providers

The access handler uses a provider framework which allows new services to be integrated easily by implementing a specific interface.

### Adding a provider

Providers are defined in accesshandler/pkg/providers if this provider is for a service which may have multiple possible providers, create a parent folder and
then create a specific folder for your new provider. E.g aws/sso or azure/ad otherwise, a single folder is ok e.g okta

One of the simplest ways to get started is to copy an existing provider and update the api calls.
A provider should have the following files.

```
accesshandler
 ┗ pkg
   ┗ providers
     ┗ okta
       ┣ 📜 access.go
       ┣ 📜 errors.go
       ┣ 📜 okta_test.go
       ┣ 📜 okta.go
       ┣ 📜 options.go
       ┗ 📜 validate.go
```

### access.go

Should contain an implementation for the accessor interface.

```go
type Accessor interface {
	// Grant the access.
	Grant(ctx context.Context, subject string, args []byte) error

	// Revoke the access.
	Revoke(ctx context.Context, subject string, args []byte) error
}
```

> It should also contain a struct `Args` always call this `Args` by convention.

Reviewer: ```suggestion
This file should also contain an `Args` struct. By convention, we always call this `Args`.
```
